packages/vector_graphics/lib/src/vector_graphics.dart

```diff
@@ -570,6 +570,29 @@ class _RawPictureVectorGraphicWidget extends SingleChildRenderObjectWidget {
 class VectorGraphicUtilities {
   const VectorGraphicUtilities._();
 
+  /// A future that completes when any in-flight vector graphic decodes have
+  /// completed.
+  ///
+  /// A vector graphic may require asynchronous work during decoding, for
+  /// example to decode an image that was embedded in the source graphic. This
+  /// method may be useful in golden image unit tests.
+  ///
+  /// ```dart
+  /// await tester.pumpWidget(MyWidgetThatHasVectorGraphics());
+  /// await tester.runAsync(() async { await vg.waitForPendingDecodes(); });
+  /// await expect(
+  ///   find.byType(MyWidgetThatHasVectorGraphics),
+  ///   matchesGoldenFile('golden_file'),
+  /// );
+  /// ```
+  ///
+  /// Without the `waitForPendingDecodes` call,
+  @visibleForTesting
+  Future<void> waitForPendingDecodes() {
```

Comment:
Can we make this debug mode only by making it throw otherwise, or do we need this for profile/release mode too?

Reply:
I think that works until someone asks for this in integration tests, but it's probably not as much of an issue there...?

Reply:
No fake async zones in integration tests anyway

Reply:
Also it's currently broken for integration tests anyway, so might as well fail loudly.